### PR TITLE
[Ukrainian] Translate `/advanced/structuring.md`

### PR DESCRIPTION
**Pull request by Nazar-Ant**

### CODE REVIEW

File: site/docs/uk/advanced/structuring.md [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+const bot = new Bot("<token>");
+
+bot.use(lists);
+// ... можливо, тут буде більше модулів на кшталт `todo`
+
+bot.start();
+```
+
+За потреби ви можете використовувати [плагін для маршрутизації](../plugins/router.md) або [`bot.route`](https://deno.land/x/grammy/mod.ts?s=Composer#method_route_0) для об'єднання різних модулів, якщо ви можете заздалегідь визначити, який middleware відповідальний за це.
```
Comment (deptyped) [inline site/docs/uk/advanced/structuring.md:68]:
```suggestion
За потреби ви можете використовувати [плагін для маршрутизації](../plugins/router.md) або [`bot.route`](https://deno.land/x/grammy/mod.ts?s=Composer#method_route_0) для об'єднання різних модулів, якщо ви можете заздалегідь визначити який модуль повинен обробити оновлення.
```

File: site/docs/uk/advanced/structuring.md [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+Однак пам'ятайте, що дуже важко сказати, як саме потрібно структурувати бота в загальних рисах.
+Як зазвичай у програмному забезпеченні, робіть це так, щоб це мало найбільший сенс :wink:
+
+## Визначення типів для вилученого middleware
```
Comment (deptyped) [inline site/docs/uk/advanced/structuring.md:73]:
```suggestion
## Визначення типів для окремого middleware
```

File: site/docs/uk/advanced/structuring.md [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+
+Описана вище структура з використанням `Composer` працює добре.
+Однак іноді ви можете опинитися в ситуації, коли ви хочете вилучити обробник у функцію, замість того, щоб створювати новий `Composer` і додавати до нього логіку.
+Це вимагає додавання правильних визначень типів middleware до ваших обробників, оскільки вони більше не можуть бути встановлені за допомогою `Composer`.
```
Comment (deptyped) [inline site/docs/uk/advanced/structuring.md:77]:
```suggestion
Це вимагає додавання правильних визначень типів до ваших обробників, оскільки вони більше не можуть бути виведені через `Composer`.
```

File: site/docs/uk/advanced/structuring.md [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+
+grammY експортує визначення типів для всіх **звужених типів middleware**, таких як middleware, яке ви можете передати обробникам команд.
+Крім того, він експортує визначення типів для **звужених обʼєктів контексту**, які використовуються у цьому middleware.
+Обидва типи параметризуються за допомогою вашого [власного обʼєкта контексту](../guide/context.md#налаштування-об-єкта-контексту).
+Отже, обробник команд матиме тип `CommandMiddleware<MyContext>` та обʼєкт контексту `CommandContext<MyContext>`.
+Їх можна використовувати наступним чином.
```
Comment (deptyped) [inline site/docs/uk/advanced/structuring.md:83]:
```suggestion
grammY експортує визначення типів для всіх **звужених типів middleware**, наприклад, для middleware, які ви можете передати обробникам команд.
Крім того, він експортує визначення типів для **звужених обʼєктів контексту**, які використовуються у таких middleware.
Обидва типи параметризуються за допомогою вашого [власного обʼєкта контексту](../guide/context.md#налаштування-об-єкта-контексту).
Відповідно, обробник команди матиме тип `CommandMiddleware<MyContext>`, а обʼєкт контексту матиме тип `CommandContext<MyContext>`.
Їх можна використовувати наступним чином.
```

**Review by deptyped — APPROVED** [review]

File: site/docs/uk/advanced/structuring.md [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+
+const bot = new Bot("<token>");
+
+bot.use(lists);
+// ... можливо, тут буде більше модулів на кшталт `todo`
+
+bot.start();
+```
+
+За потреби ви можете використовувати [плагін для маршрутизації](../plugins/router.md) або [`bot.route`](https://deno.land/x/grammy/mod.ts?s=Composer#method_route_0) для об'єднання різних модулів, якщо ви можете заздалегідь визначити, який модуль повинен обробити оновлення.
```
Comment (LWJerri) [inline site/docs/uk/advanced/structuring.md:68]:
```suggestion
За потреби ви можете використовувати [плагін для маршрутизації](../plugins/router.md) або [`bot.route`](https://deno.land/x/grammy/mod.ts?s=Composer#method_route_0) для обʼєднання різних модулів, якщо ви можете заздалегідь визначити, який модуль повинен обробити оновлення.
```

File: site/docs/uk/advanced/structuring.md [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+> grammY ще досить молодий і поки що не надає жодних офіційних інтеграцій з DI-контейнерами.
+> Підпишіться на [@grammyjs_news](https://t.me/grammyjs_news), щоб отримати сповіщення, як тільки ми надамо підтримку цього.
+
+Ви можете структурувати свій код як завгодно, універсального рішення не існує.
+З огляду на це, проста і перевірена стратегія структурування вашого коду полягає в наступному.
+
+1. Групуйте семантично повʼязані речі в одному файлі або каталозі, в залежності від кількості коду.
+   Кожен із таких блоків має надавати middleware, який оброблятиме певні повідомлення.
+2. Створюйте екземпляр бота в єдиному місці, в якому також буде встановлено всі middleware, які він використовує.
+3. **Необов'язково**. Попередньо фільтруйте оновлення та передавайте їх далі за правильним маршрутом.
```
Comment (LWJerri) [inline site/docs/uk/advanced/structuring.md:22]:
```suggestion
3. **Необовʼязково**. Попередньо фільтруйте оновлення та передавайте їх далі за правильним маршрутом.
```

**Review by LWJerri — APPROVED** [review]

**Comment by KnorpelSenf** [comment]
It seems like #705 wasn't synced, right?

**Comment by Nazar-Ant** [comment]
@KnorpelSenf Because these changes affect many files that have not yet been reviewed, it would be difficult to apply these changes to every translation. I decided that we would gradually apply these changes during the review process and, after translation, apply them to the pages that have already been fully translated.

**Pull request merged**